File: CODIGO 119/8.ts
function generateBrainfuckCode(N: number, f0: number, f1: number): string {
    let bfCode = '';

    // Inicializar celdas de memoria para f0 y f1
    bfCode += '[-]'; // Inicializar la celda de memoria en 0
    bfCode += '+'.repeat(f0);
    bfCode += '>'.repeat(f1);

    // Generar secuencia Fibonacci
    bfCode += '[';
    bfCode += '->+[-<+]';
    bfCode += '>'.repeat(f1);
    bfCode += '<<<<<-]';
    bfCode += '[' + '+'.repeat(f1) + '>'.repeat(f0) + '<-]>[->+<][>+<-]<[<+>>+<-]>[->+<]';

    // Imprimir N valores de la secuencia
    bfCode += '>'.repeat(N - 1);
    bfCode += '[';
    bfCode += '>++++++[<++++++++>-]<.[-]<[-]<';
    bfCode += '<'.repeat(N - 1);
    bfCode += '-]';

    return bfCode;
}

const N = 5;
const f0 = 1;
const f1 = 3;

const bfCode = generateBrainfuckCode(N, f0, f1);
console.log(bfCode);

File: CODIGO 119/8.dart
package main

import (
	"fmt"
	"strings"
)

func main() {
	N := 5
	f0 := 1
	f1 := 3

	bfCode := generateBrainfuckCode(N, f0, f1)
	fmt.Println(bfCode)
}

func generateBrainfuckCode(N, f0, f1 int) string {
	var sb strings.Builder

	// Inicializar celdas de memoria para f0 y f1
	sb.WriteString("[-]") // Inicializar la celda de memoria en 0
	sb.WriteString(fmt.Sprintf("+%s", strings.Repeat("+", f0)))
	sb.WriteString(fmt.Sprintf(">%s", strings.Repeat("+", f1)))

	// Generar secuencia Fibonacci
	sb.WriteString("[")
	sb.WriteString(fmt.Sprintf->+[-<+>]"))
	sb.WriteString(fmt.Sprintf(">>%s", strings.Repeat("+", f1)))
	sb.WriteString("<<<<<-]")
	sb.WriteString(fmt.Sprintf("[%s>%s<-]>[->+<]", strings.Repeat("+", f1), strings.Repeat("+", f0)))
	sb.WriteString("[>+<-]<[<+>>+<-]>[->+<]")

	// Imprimir N valores de la secuencia
	sb.WriteString(fmt.Sprintf(">%s", strings.Repeat(">", N-1)))
	sb.WriteString("[")
	sb.WriteString(">++++++[<++++++++>-]<.[-]<[-]<")
	sb.WriteString(fmt.Sprintf(">%s", strings.Repeat("<", N-1)))
	sb.WriteString("-]")

	return sb.String()
}
